### PR TITLE
VEN-1311 | Query pier services on areas

### PR DESCRIPTION
**Pull request by a-rmz**
## Description :sparkles:
* Add annotated fields for the pier/section services (e.g. electricity, water, gate) on the harbor/area
  * Expose the fields on the GQL nodes
* Remove the `repairArea` from `winter storage sections` as it won't be used anymore

### Closes :no_good_woman:
**[VEN-1311](https://helsinkisolutionoffice.atlassian.net/browse/VEN-1311):** Allow querying for services in harbors and winterStorageAreas

### Automated tests :gear:️
```shell
$ pytest resources/tests/test_resources_queries.py::test_get_harbor_services
$ pytest resources/tests/test_resources_queries.py::test_get_winter_storage_area_services
```

### Manual testing :construction_worker_man:
```graphql
query Harbor {
  harbor(id: "SGFyYm9yTm9kZTowMDkyYWU3MC1jYzQ2LTQ0ZGYtODU3ZS03OWVhMzRlNTczZDU=") {
    id
    properties {
      mooring
      electricity
      water
      wasteCollection
      gate
      lighting
      suitableBoatTypes {
        id
        name
      }
    }
  }
}
```

Expected result:
```json
{
  "data": {
    "harbor": {
      "id": "SGFyYm9yTm9kZTowMDkyYWU3MC1jYzQ2LTQ0ZGYtODU3ZS03OWVhMzRlNTczZDU=",
      "properties": {
        "name": "Kipparlahden venesatama",
        "mooring": true,
        "electricity": false,
        "water": false,
        "wasteCollection": true,
        "gate": true,
        "lighting": false,
        "suitableBoatTypes": [
          {
            "id": "2",
            "name": "Rowing boat"
          },
          {
            "id": "3",
            "name": "Outboard motor boat"
          },
          {
            "id": "4",
            "name": "Inboard motor boat"
          },
          {
            "id": "5",
            "name": "Mid-engined boat"
          },
          {
            "id": "6",
            "name": "Sailing boat / motor-sailing boat"
          }
        ]
      }
    }
  }
}
```

### CODE REVIEW

File: resources/tests/test_resources_queries.py [hunk 1]
```diff
@@ -1267,3 +1268,139 @@ def test_berth_is_invoiceable(api_client):
...
+        suitable_boat_types__count=2,
+    )
+    harbor = Harbor.objects.get(id=first_pier.harbor.id)
+
+    combined_types = list(
+        first_pier.suitable_boat_types.values_list("id", flat=True)
+    ) + list(second_pier.suitable_boat_types.values_list("id", flat=True))
+
+    suitable_boat_types = [
+        {"id": str(type_id)} for type_id in sorted(set(combined_types))
```
Comment (haavikko) [inline resources/tests/test_resources_queries.py:1300]: Does the backend always return the results in sorted order?

**Review by haavikko — APPROVED** [review]
Looks good to me, before merging please check the comment in tests

File: resources/tests/test_resources_queries.py (hunk 1 above)
Comment (a-rmz) [reply resources/tests/test_resources_queries.py:1300]: It does:
```python
type_ids = (
    self.piers.order_by("suitable_boat_types__id")
    .distinct("suitable_boat_types__id")
    .values_list("suitable_boat_types", flat=True)
)
```

File: resources/models.py [hunk 2]
```diff
@@ -169,6 +170,32 @@ def get_queryset(self):
                 number_of_places=Subquery(
                     number_of_places_qs, output_field=SmallIntegerField()
                 ),
+                electricity=BoolOr("piers__electricity"),
```
[the remark this answers is not in the document]
Comment (a-rmz) [reply resources/models.py:173]: Same here 🙌  it's a [Postgres specific function](https://docs.djangoproject.com/en/3.2/ref/contrib/postgres/aggregates/) but very nice to have it.

**Pull request merged**